Convert docs introduction to mdx

Yofamb opened the pull request:
## Description

Make the Introduction page of the docs an MDX file.

## Related Issues

Addresses #17989 


## before
![screencapture-gatsbyjs-org-docs-2019-10-14-09_29_06](https://user-images.githubusercontent.com/3781418/66723495-8f03a880-ee65-11e9-9b47-ac18634e2fa5.png)

## after
![screencapture-localhost-5500-docs-2019-10-14-09_28_30](https://user-images.githubusercontent.com/3781418/66723497-90cd6c00-ee65-11e9-997f-d9b240235291.png)

tesseralis commented on www/src/pages/docs/index.mdx:
```diff
@@ -0,0 +1,65 @@
+
+There are two main ways to get started with Gatsby:
+
+1. <Link to="/tutorial/">Tutorial</Link>: Step-by-step instructions on how to install Gatsby and start a project: written for people without Gatsby or web development experience, though these learning resources have helped developers of all skill levels.
```
You can just do Markdown links:

```
[Tutorial](/tutorial)
```

tesseralis commented on www/src/pages/docs/index.mdx:
```diff
@@ -0,0 +1,65 @@
+import { Link } from "gatsby"
```
This file should be `/docs/docs/index.md`.

tesseralis commented on www/src/pages/docs/index.mdx:
```diff
@@ -0,0 +1,65 @@
+
+Visit the <Link to="/contributing">Contributing</Link> section to find guides on the Gatsby community, code of conduct, and how to get started contributing to Gatsby.
+
+export default ({ children, location }) => (
```
You shouldn't need to do all this: the only thing component you'd need is `<EmailCaptureForm>`. Look at `template-doc-markdown` -- the content here is mostly the same as that, except for the email submission form.

Yofamb replied:
Cheers, i'll have a look over the weekend. Thanks for the feedback

Yofamb replied:
Hi @tesseralis I've fixed up the pull request. I still wasn't sure the best way to set up the email capture form. What's your thoughts on how i've achieved this?

Yofamb commented on www/src/templates/template-docs-markdown.js:
```diff
@@ -169,6 +173,7 @@ export const pageQuery = graphql`
         title
         overview
         issue
+        showSignupForm
```
potentially this should be called `enabledSignupForm` ?

tesseralis commented on www/src/templates/template-docs-markdown.js:
```diff
@@ -136,6 +137,9 @@ function DocsTemplate({ data, location, pageContext: { next, prev } }) {
                     See the issue relating to this stub on GitHub
                   </a>
                 )}
+                {page.frontmatter.showSignupForm && (
```
You should be able to just import and use the <EmailCaptureFrom> in [MDX](https://www.gatsbyjs.org/docs/mdx/writing-pages/).

In `index.md`:

```md
import EmailCaptureForm from 'www/src/components/EmailCaptureForm'

...content...

<EmailCaptureForm signupMessage="..."/>
```

Yofamb commented:
@tesseralis I've updated everything now. Not sure why I made everything hard for myself at the start! this was a really simple issue. 

I am getting one test fail but i'm unsure why this is happening - could you please take a look and let me know what may be wrong? (it seems like un unrelated failure?). Once you've given the thumbs up ill mark this PR as ready to review.

tesseralis commented:
@yofamb thanks! Next, could you verify that the page look the same before and after the change? Build the website locally and post a screenshot of the page to make sure it works.

Yofamb commented:
Hi @tesseralis I've rebased the pull request, added screenshots and have compared the content from live docs and my branch. 

The only thing that i've noticed is different is what gets added to the title tag. See attached image Could you please review when you have the time. Cheers!

<img width="490" alt="Screen Shot 2019-10-16 at 10 01 52 am" src="https://user-images.githubusercontent.com/3781418/66876034-0c612180-effc-11e9-83ab-edff3f8e1af3.png">

tesseralis commented:
@yofamb this looks great! I'll ping in the rest of the gatsby team to see what to do about the title tag.

tesseralis commented on docs/docs/index.md:
```diff
@@ -0,0 +1,34 @@
+---
+title: Gatsby.js Documentation
```
You need to add `disableTableOfContents: true` to disable the right sidebar.

marcysutton commented:
An update for the HTML title: I think it's fine to merge as-is. We can always update it later.
